src/server/pps/server/api_server.go

```diff
@@ -712,6 +712,37 @@ func (a *apiServer) ListJob(ctx context.Context, request *ppsclient.ListJobReque
 	}, nil
 }
 
+func (a *apiServer) DeleteJob(ctx context.Context, request *ppsclient.DeleteJobRequest) (response *google_protobuf.Empty, err error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "DeleteJob")
+	defer func(start time.Time) { metricsFn(start, err) }(time.Now())
+
+	pfsAPIClient, err := a.getPfsClient()
+	if err != nil {
+		return nil, err
+	}
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return nil, err
+	}
+	jobInfo, err := persistClient.InspectJob(ctx, &ppsclient.InspectJobRequest{
+		Job: request.Job,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := a.deleteJob(ctx, jobInfo); err != nil {
+		return nil, err
+	}
+	if _, err := pfsAPIClient.DeleteRepo(ctx, &pfsclient.DeleteRepoRequest{Repo: jobInfo.OutputCommit.Repo}); err != nil {
+		return nil, err
+	}
```

Comment:
Good point, I think if you delete a job from a pipeline that's already run then nothing will happen immediately because `runPipeline` will have already seen the relevant commits and won't be listening for them anymore. However, if `runPipeline` were to get restarted for this pipeline (due to a node failure, or a pipeline update) then it would recreate the job when it received those commits. Not particularly sane behavior. What would you think of just erroring if the user attempts to delete such a job and saying something like: `"job foo can't be delete because it's part of pipeline bar, delete pipeline bar instead"`?